src/dmd/toobj.d

```diff
@@ -449,6 +449,12 @@ void toObjFile(Dsymbol ds, bool multiobj)
             // Put out the members
             id.members.foreachDsymbol( (s) { visitNoMultiObj(s); } );
 
+            // Objetive-C protocols are only outputted if they're implemented by
+            // a class. If they are implemented by a class, they're outputted
+            // via the class declaration.
```

Comment:
This seems strange. What does "via the class declaration" mean?

Reply:
The class symbol contains a reference to the protocol symbol. When the class symbol is output, it will automatically output the protocol symbol as well. If the protocol is never implemented, it won't be output.